### PR TITLE
added the softshrink support for mindspore

### LINKED ISSUE #21505: soft shrink

### PR DESCRIPTION
**Pull request by sushmanthreddy**
close #21505

### CODE REVIEW

**Comment by sushmanthreddy** [comment]
@mohame54 can u pls review the pull request??

File: ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py [hunk 1]
```diff
@@ -360,3 +360,37 @@
...
+#     helpers.test_frontend_function(
+#         input_dtypes=input_dtype,
+#         backend_to_test=backend_fw,
+#         frontend=frontend,
+#         test_flags=test_flags,
+#         fn_tree=fn_tree,
+#         on_device=on_device,
+#         x=x[0],
+#         lambd=lambd,
+#     )
```
Comment (a0m0rajab) [inline ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py:396]: why did you comment this out?

Comment (sushmanthreddy) [reply ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py:396]: @a0m0rajab there is no backend support yet for mindspore... 
So we have commented out the code

**Comment by sushmanthreddy** [comment]
@a0m0rajab can u plss review this once ,have removed the merge conflicts
..

once pls review it at least give a review..

**Comment by sushmanthreddy** [comment]
@mohame54 @hirwa-nshuti can u pls review this once?

**Comment by sushmanthreddy** [comment]
@DragosStoican can u pls review this pull request??
it has been three days there is no activity..it would be great help,if anyone can review this

File: ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py [hunk 2]
```diff
@@ -424,6 +424,40 @@
...
 #     )
 
+# def _filter_dtypes(input_dtype):
+#     assume(("bfloat16" not in input_dtype) and ("float16" not in input_dtype))
+
+# # softshrink
+# @handle_frontend_test(
+#     fn_tree="mindspore.ops.function.nn_func.softshrink",
+#     dtype_and_input=helpers.dtype_and_values(
+#         available_dtypes=helpers.get_dtypes("float"),
```
Comment (DragosStoican) [inline ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py:434]: We should be testing for all valid dtypes by using the "valid" argument here. And in that case I think there is no need for _filter_dtypes anymore either

**Comment by sushmanthreddy** [comment]
@DragosStoican it has been updated to as required mentions..can u pls review now??

**Pull request merged**